# Baseline Model Training

This notebook will be used to train and evaluete initial machine-learning models for the Board Game Rating Predictor project.

Goal is to create simple baseline models that predict `Rating Average` using the prepared modelling datasets created during feature engineering.

This notebook will:

- load prepared training and testing datasets
- train simple baseline model
- train initial regression model
- evaluate model performance
- compare model results
- document baseline modelling decisions

In [7]:
# Import libraries used throghout this notebook
import pandas as pd

# Import Path for creating file paths that work
# reliably across different operating systems.
from pathlib import Path

In [8]:
# Define current notebook folder and project root
current_folder = Path.cwd()
project_root = current_folder.parent

print("Current folder;", current_folder)
print("Project root:", project_root)

Current folder; c:\Users\Zombie\Desktop\vscode-projects\board-game-rating-predictor\notebooks
Project root: c:\Users\Zombie\Desktop\vscode-projects\board-game-rating-predictor


## Load Prepared Modelling Datasets

The prepared training and testing datasets are loaded from the `data/processed` folder.

These files were created during the feature engineering notebook.

Loaded files are following:
- training features
- testing features
- training target values
- testing target values
- model feature names
- imputation summary

Because the feature files were already prepared and imputed, they should not contain missing values.

In [9]:
# Define paths to prepared modelling dataset files
processed_data_folder = project_root / "data" / "processed"

x_train_path = processed_data_folder / "x_train_prepared.csv"
x_test_path = processed_data_folder / "x_test_prepared.csv"
y_train_path = processed_data_folder / "y_train.csv"
y_test_path = processed_data_folder / "y_test.csv"

model_feature_names_path = processed_data_folder / "model_feature_names.csv"
imputation_summary_path = processed_data_folder / "imputation_summary.csv"

print("Processed data folder:", processed_data_folder)
print("Folder exists:", processed_data_folder.exists())

Processed data folder: c:\Users\Zombie\Desktop\vscode-projects\board-game-rating-predictor\data\processed
Folder exists: True


In [10]:
# Cofrim that all required prepared modelling files exist
prepared_file_paths = [
    x_train_path,
    x_test_path,
    y_train_path,
    y_test_path,
    model_feature_names_path,
    imputation_summary_path
]

for file_path in prepared_file_paths:
    print(file_path.name, "exists:", file_path.exists())

x_train_prepared.csv exists: True
x_test_prepared.csv exists: True
y_train.csv exists: True
y_test.csv exists: True
model_feature_names.csv exists: True
imputation_summary.csv exists: True


In [11]:
# Load prepared modelling datasets
X_train = pd.read_csv(x_train_path)
X_test = pd.read_csv(x_test_path)

y_train_data = pd.read_csv(y_train_path)
y_test_data = pd.read_csv(y_test_path)

model_feature_names = pd.read_csv(model_feature_names_path)
imputation_summary = pd.read_csv(imputation_summary_path)

print("Prepared modelling files loaded successfully.")

Prepared modelling files loaded successfully.


In [12]:
# Define target colum
target_column = "Rating Average"

# Convert target DataFrames into Series for modelling
y_train = y_train_data[target_column]
y_test = y_test_data[target_column]

print("y_train type:", type(y_train))
print("y_test type:", type(y_test))

y_train type: <class 'pandas.Series'>
y_test type: <class 'pandas.Series'>


In [13]:
# Check the shapes of the loaded modelling datasets
print("X_train shape:", X_train.shape)
print("X_test shape:", X_test.shape)
print("y_train shape:", y_train.shape)
print("y_test shape:", y_test.shape)
print("model_feature_names shape:", model_feature_names.shape)
print("imputation_summary shape:", imputation_summary.shape)

X_train shape: (16273, 35)
X_test shape: (4069, 35)
y_train shape: (16273,)
y_test shape: (4069,)
model_feature_names shape: (35, 1)
imputation_summary shape: (6, 2)


In [14]:
# Verify if loaded feature columns match saved modl feature names
expected_feature_names = model_feature_names["Feature"].tolist()

x_train_columns_match = X_train.columns.tolist() == expected_feature_names
x_test_columns_match = X_test.columns.tolist() == expected_feature_names

print("X_train columns match saved feature names:", x_train_columns_match)
print("X_test columns match saved feature names:", x_test_columns_match)

X_train columns match saved feature names: True
X_test columns match saved feature names: True


In [15]:
# Check missing values in the loaded prepared datasets
print("Missing values in X_train:", X_train.isnull().sum().sum())
print("Missing values in X_test:", X_test.isnull().sum().sum())
print("Missing values in y_train:", y_train.isnull().sum())
print("Missing values in y_test:", y_test.isnull().sum())

Missing values in X_train: 0
Missing values in X_test: 0
Missing values in y_train: 0
Missing values in y_test: 0


In [16]:
# Preview loaded training features
X_train.head()

,Year Published,Min Players,Min Age,Complexity Average,Max Players Log,Play Time Log,Year Published Missing,Min Players Missing,Max Players Missing,Play Time Missing,...,Mechanic: Area Movement,Mechanic: Simultaneous Action Selection,Domain: Wargames,Domain: Strategy Games,Domain: Family Games,Domain: Thematic Games,Domain: Abstract Games,Domain: Children's Games,Domain: Party Games,Domain: Customizable Games
0,2017.0,2.0,10.0,2.00,1.791759,4.110874,0,0,0,0,...,0,0,0,0,0,0,0,0,0,0
1,2013.0,2.0,11.0,1.43,1.945910,3.828641,0,0,0,0,...,0,0,0,0,0,0,0,0,0,0
2,2019.0,2.0,8.0,2.00,1.098612,3.433987,0,0,0,0,...,0,0,0,0,0,0,0,0,0,0
3,2004.0,2.0,12.0,3.00,1.098612,7.090910,0,0,0,0,...,1,0,1,0,0,0,0,0,0,0
4,2010.0,2.0,8.0,1.40,1.945910,2.772589,0,0,0,0,...,0,0,0,0,0,0,0,0,0,0


In [17]:
# Preview the loaded training target values
y_train.head()

0    5.86
1    4.53
2    7.33
3    7.60
4    5.97
Name: Rating Average, dtype: float64

### Prepared Modelling Dataset Load Result

Prepared modelling datasets were loaded successfully from the `data/processed` folder.

Training feature dataset contains 16,273 rows and 35 feature columns.

Testing feature dataset contains 4,069 rows and 35 feature columns.

Target values were loaded and converted into one-dimensional Series objects for modelling.

Loaded feature columns match the saved model feature names.

Loaded prepared datasets contain no missing values and and ready for baseline model training.